Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp

```diff
@@ -181,17 +181,20 @@ void RemoteImageBufferSet::prepareBufferForDisplay(const WebCore::Region& dirtyR
         return;
     }
 
-    IntRect layerBounds = IntRect { { }, expandedIntSize(m_logicalSize) };
+    FloatRect bufferBounds { { }, m_logicalSize };
 
     GraphicsContext& context = m_frontBuffer->context();
     context.resetClip();
 
-    if (m_previousFrontBuffer && m_frontBuffer != m_previousFrontBuffer && !dirtyRegion.contains(layerBounds)) {
-        Region copyRegion(m_previouslyPaintedRect ? *m_previouslyPaintedRect : layerBounds);
-        copyRegion.subtract(dirtyRegion);
-        IntRect copyRect = copyRegion.bounds();
-        if (!copyRect.isEmpty())
-            m_frontBuffer->context().drawImageBuffer(*m_previousFrontBuffer, copyRect, copyRect, { CompositeOperator::Copy });
+    if (m_previousFrontBuffer && m_frontBuffer != m_previousFrontBuffer) {
+        IntRect enclosingCopyRect { m_previouslyPaintedRect ? *m_previouslyPaintedRect : enclosingIntRect(bufferBounds) };
+        if (!dirtyRegion.contains(enclosingCopyRect)) {
+            Region copyRegion(enclosingCopyRect);
+            copyRegion.subtract(dirtyRegion);
```

Comment:
Don't you need to fix this same bug for in-process backing store? (Until Matt figures out how to share the code).

Reply:
LOL I see he said the same thing, ignore me.